packages/backend/src/ee/database/migrations/20250709140139_ai_artifacts_and_versions.ts

```diff
@@ -0,0 +1,131 @@
+            .inTable(AiArtifactsTable)
+            .onDelete('CASCADE');
+        table
+            .uuid('ai_prompt_uuid')
+            .nullable()
+            .references('ai_prompt_uuid')
+            .inTable(AiPromptTable)
+            .onDelete('SET NULL');
+        table.timestamp('created_at').notNullable().defaultTo(knex.fn.now());
+        table
+            .uuid('created_by_user_uuid')
+            .nullable()
+            .references('user_uuid')
+            .inTable(UsersTable)
+            .onDelete('SET NULL');
+        table.integer('version_number').notNullable();
+        table.jsonb('viz_config_output').nullable();
+        table.jsonb('filters_output').nullable();
+        table.jsonb('metric_query').nullable();
+
+        table.unique(['ai_artifact_uuid', 'version_number']);
+
+        table.index('ai_artifact_uuid');
+        table.index('ai_prompt_uuid');
+        table.index('created_by_user_uuid');
+        table.index('created_at');
+    });
+
+    /*
+    Could we do this? 
+    
+    1. Each ai_prompt with viz_config_output could become its own artifact
+    2. Create artifacts for prompts with viz configs:
+    
+    const promptsWithVizConfig = await knex('ai_prompt')
+        .select(['ai_prompt_uuid', 'ai_thread_uuid', 'created_by_user_uuid', 
+                 'created_at', 'responded_at', 'viz_config_output', 
+                 'filters_output', 'metric_query', 'saved_query_uuid'])
+        .whereNotNull('viz_config_output')
+        .orderBy('created_at');
+
+    for (const prompt of promptsWithVizConfig) {
+        // Create individual artifact for each prompt
+        const [artifact] = await knex('ai_artifacts').insert({
+            ai_thread_uuid: prompt.ai_thread_uuid,
+            created_by_user_uuid: prompt.created_by_user_uuid,
+            created_at: prompt.created_at,
+            artifact_type: 'chart',
+            title: extractTitleFromVizConfig(prompt.viz_config_output), // Custom logic
+            description: extractDescriptionFromVizConfig(prompt.viz_config_output),
+            saved_query_uuid: prompt.saved_query_uuid,
+        }).returning('ai_artifact_uuid');
+
+        // Create initial version
+        await knex('ai_artifact_versions').insert({
+            ai_artifact_uuid: artifact.ai_artifact_uuid,
+            ai_prompt_uuid: prompt.ai_prompt_uuid,
+            created_at: prompt.responded_at || prompt.created_at,
+            created_by_user_uuid: prompt.created_by_user_uuid,
+            version_number: 1,
+            viz_config_output: prompt.viz_config_output,
+            filters_output: prompt.filters_output,
+            metric_query: prompt.metric_query,
+        });
+    }
+
+    3. After migration is complete and verified:
+       - Update application code to use artifacts instead of ai_prompt.viz_config_output
+       - Eventually drop viz_config_output, filters_output, metric_query from ai_prompt
+       - Remove backwards compatibility code
```

Comment:
note ^

Reply:
do we need to support both flows where we read from prompt and from artifact? no right? once we move to the new table, that's it 🤔